### PR TITLE
Add launcher example

### PR DESCRIPTION
**Pull request by fcollonval**
@jtpio I took the liberty to copy your excellent https://github.com/jtpio/jupyterlab-python-file extension. Is it ok?

### CODE REVIEW

**Comment by jtpio** [comment]
Absolutely!

It's a great fit as a launcher example. And it can easily be modified to create other types of files too.

File: launcher/README.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+Then you can use it in the extension by importing it:
+
+```ts
+// src/index.ts#L10-L10
+
+import { ILauncher } from '@jupyterlab/launcher';
+```
+
+And finally you can request it as extension dependency:
```
Comment (jtpio) [inline launcher/README.md]: Or?

```suggestion
And finally you can add it to the list of dependencies:
```

File: launcher/README.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+```ts
+// src/index.ts#L22-L33
+
+const extension: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlab-examples/launcher',
+  autoStart: true,
+  optional: [ILauncher, IMainMenu, ICommandPalette],
+  requires: [IFileBrowserFactory],
```
Comment (jtpio) [inline launcher/README.md]: Should we position `requires` first, and then `optional`? So the order is the same as the parameters in the `activate` function.

Comment (jtpio) [reply launcher/README.md]: Not sure there is a clear pattern for that though (also in core lab).

File: launcher/README.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  activate: (
+    app: JupyterFrontEnd,
+    browserFactory: IFileBrowserFactory,
+    launcher: ILauncher | null,
+    menu: IMainMenu | null,
+    palette: ICommandPalette | null
+  ) => {
+```
+
+In this example, the `ILauncher` interface is requested as optional dependency and not as classical dependency. This allow other application without launcher to be able
```
Comment (jtpio) [inline launcher/README.md]: We could directly use the same term with `require`:

```suggestion
In this example, the `ILauncher` interface is requested as optional dependency and not as a required dependency. This lets other applications without a launcher be able
```

File: launcher/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    if (palette) {
+      palette.addItem({
+        command: CommandIDs.createNew,
+        args: { isPalette: true },
+        category: PALETTE_CATEGORY
+      });
+    }
+
+    // Add the command to the menu
+    if (menu) {
```
Comment (jtpio) [inline launcher/src/index.ts]: Thanks for adding this `if`, I'll update that in the upstream repo.

**Comment by jtpio** [comment]
Thanks @fcollonval!

I did a small update for the extension in the following PR: https://github.com/jtpio/jupyterlab-python-file/pull/6
Maybe we want to apply some of these changes here too?

File: launcher/README.md (hunk 2 above)
Comment (fcollonval) [reply launcher/README.md]: I agree with you. There is no clear pattern to have the attribute in alphabetical order. And as the required arguments come before the optional ones, it makes sense to list them in that order.

**Comment by fcollonval** [comment]
Thanks @jtpio for the review and to agree on reusing your extension. I added your improvement from the mentioned PR.

**Pull request merged**